pandas/_libs/algos.pyx

```diff
@@ -71,6 +76,42 @@ class NegInfinity(object):
     __ge__ = lambda self, other: isinstance(other, NegInfinity)
 
 
+cpdef ndarray[int64_t, ndim=1] unique_deltas(ndarray[int64_t] arr):
```

Comment:
as an aside I think we usually call this diff elsewhere, so maybe can share code, and might thing about renaming (future PR to think about)